### PR TITLE
prov/sockets: reset rx_ctx->progress_start when an entry is removed

### PR DESCRIPTION
**Pull request by zhenggb72**
Signed-off-by: Gengbin Zheng <gengbin.zheng@intel.com>

### CODE REVIEW

File: prov/sockets/src/sock_progress.c [hunk 1]
```diff
@@ -1169,6 +1169,7 @@ ssize_t sock_rx_claim_recv(struct sock_rx_ctx *rx_ctx, void *context,
 
 		dlist_remove(&rx_buffered->entry);
 		sock_rx_release_entry(rx_buffered);
+		rx_ctx->progress_start = &rx_ctx->rx_buffered_list;
```
Comment (shefty) [inline prov/sockets/src/sock_progress.c:1172]: Can you please add more detail to the commit message as to why this change is needed and what it fixes?

Comment (zhenggb72) [reply prov/sockets/src/sock_progress.c]: added some comments

Comment (zhenggb72) [reply prov/sockets/src/sock_progress.c]: It fixed seg fault in a multi-threading test.

**Review by shefty — APPROVED** [review]
